test/e2e/operator_expectations.go

```diff
@@ -608,3 +613,481 @@ func (tc *testConfig) ExpectAutoscalerScalesOut() error {
 		return len(nodeList.Items) == clusterInitialTotalNodes, nil
 	})
 }
+
+func (tc *testConfig) ExpectNodeToBeDrainedBeforeDeletingMachine() error {
+	listOptions := client.ListOptions{
+		Namespace: namespace,
+	}
+
+	var machine mapiv1beta1.Machine
+	var nodeName string
+	var node *corev1.Node
+
+	glog.Info("Get machineList with at least one machine with NodeRef set")
+	if err := wait.PollImmediate(1*time.Second, waitShort, func() (bool, error) {
+		machineList := mapiv1beta1.MachineList{}
+		if err := tc.client.List(context.TODO(), &listOptions, &machineList); err != nil {
+			glog.Errorf("error querying api for machineList object: %v, retrying...", err)
+			return false, nil
+		}
+		for _, machineItem := range machineList.Items {
+			// empty or non-worker role skipped
+			if machineItem.Labels["sigs.k8s.io/cluster-api-machine-role"] == "worker" {
+				if machineItem.Status.NodeRef != nil && machineItem.Status.NodeRef.Name != "" {
+					machine = machineItem
+					nodeName = machineItem.Status.NodeRef.Name
+					return true, nil
+				}
+			}
+		}
+		return false, fmt.Errorf("no machine found with NodeRef not set")
+	}); err != nil {
+		return err
+	}
+
+	glog.Info("Get nodeList")
+	if err := wait.PollImmediate(1*time.Second, waitShort, func() (bool, error) {
+		nodeList := corev1.NodeList{}
+		if err := tc.client.List(context.TODO(), &listOptions, &nodeList); err != nil {
+			glog.Errorf("error querying api for nodeList object: %v, retrying...", err)
+			return false, nil
+		}
+		for _, nodeItem := range nodeList.Items {
+			if nodeItem.Name == nodeName {
+				node = &nodeItem
```

Comment:
This takes the address of a local variable which seems Ok-ish here given we `break`. But it may be clearer and more future proof to use `&nodeList.Items[i]`.